### PR TITLE
❄️ Rebuild using libxml2 2.13.1 ❄️ 

### PR DESCRIPTION
**Pull request by SC426**
pdal 2.5.3 ❄️ 

**Destination channel:** Snowflake

### Links

- [PKG-5056](https://anaconda.atlassian.net/browse/PKG-5056)
- [Fix convention error with xmlErrorPtr caused by libxml2 API change](https://github.com/PDAL/PDAL/pull/4257)

### Explanation of changes:

- Update libxml2 and libcurl pinnings
- Add patch

[PKG-5056]: https://anaconda.atlassian.net/browse/PKG-5056?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PKG-4629]: https://anaconda.atlassian.net/browse/PKG-4629?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/patches/0001_Fix_xmlErrorPtr.patch [hunk 1]
```diff
@@ -0,0 +1,58 @@
+From a13d70254ef1c0788f584aeb5ec25392fe9f8497 Mon Sep 17 00:00:00 2001
+From: Steven <scrass@anaconda.com>
```
Comment (JeanChristopheMorinPerso) [inline recipe/patches/0001_Fix_xmlErrorPtr.patch]: @SC426 We should give proper attribution with patches. You can get the full patch from https://github.com/PDAL/PDAL/pull/4257.patch. Using the original patch will give us both proper attribution, authorship and commit ID. We can also go one step further and use the patch from the merged commit (https://github.com/PDAL/PDAL/commit/6320e8832e40d9a5f92af619aaaf05169226de1a.patch).

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/patches/0001_Fix_xmlErrorPtr.patch [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Date: Thu, 8 Aug 2024 10:22:57 -0600
+Subject: [PATCH] Fix convention error with xmlErrorPtr caused by libxml2 API change
```
Comment (danpetry) [inline recipe/patches/0001_Fix_xmlErrorPtr.patch]: if the libxml API changed, is there a possibility to updating to a pdal that's compatible with the new API?

**Review by danpetry — COMMENTED** [review]
I'm not sure about whether the pinnings should be as they are ideally - there are a few cbc pinnings that are overridden in host (although admittedly some are pinned loosely in cbc, so maybe we want to keep them for reproducibility) and in run, version bounds are made explicitly here whereas I think it would be better to rely on the run_exports so we don't have to manually keep up-to-date (error prone). but @skupr-anaconda made those pinnings in the first place so he may have other ideas (for example it's better to be explicit)

File: recipe/patches/0001_Fix_xmlErrorPtr.patch (hunk 1 above)
Comment (SC426) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: Done! Thanks for pointing this out.

File: recipe/patches/0001_Fix_xmlErrorPtr.patch (hunk 2 above)
Comment (SC426) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: Possibly, but someone from PSUBS would most likely need to handle that.

Comment (SC426) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: @danpetry 

```
Starting with `libxml2 2.12.0`, the API has changed the const-ness of the
`xmlError` pointers, which results in a build error due to a mismatched
type on `const xmlError *` and `xmlErrorPtr`.
This commit papers over the difference by using preprocessor conditionals.

Related libxml2 commit and issue:
- https://gitlab.gnome.org/GNOME/libxml2/-/commit/45470611b047db78106dcb2fdbd4164163c15ab7
- https://gitlab.gnome.org/GNOME/libxml2/-/commit/61034116d0a3c8b295c6137956adc3ae55720711
```

Comment (danpetry) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: oh, it's a snowflake package. ok fair

**Review by danpetry — APPROVED** [review]

**Comment by lorepirri** [comment]
@SC426 you can update `abs.yaml` to:

```
upload_channels:
  - sfe1ed40
```

(remove `channels`)

the ticket number in the description should be only one, other wise both tickets will be set to `Merged` when this one is merged.

**Review by skupr-anaconda — APPROVED** [review]
LGTM but Lorenzo's comment should be addressed

File: recipe/patches/0001_Fix_xmlErrorPtr.patch (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: It was not the best way to handle pinnings a year ago.
A plain `libxml2` in `run` is fine

Comment (danpetry) [reply recipe/patches/0001_Fix_xmlErrorPtr.patch]: > It was not the best way to handle pinnings a year ago.

yeah this method was pushed by kai IIRC

**Pull request merged**